### ISSUE TITLE
memory leak when building: cargo-geiger

### ISSUE THREAD

**Issue by Quantenzitrone** [NixOS/nixpkgs#227644]
### Steps To Reproduce

```console
nix build nixpkgs#cargo-geiger
```
It takes a while to compile, but when finished the memory usage goes really high in the testphase, 
during `Running tests/integration_tests.rs (target/x86_64-unknown-linux-gnu/release/deps/integration_tests-93b3a92f54ea3b19)` 
in the step `test test_package::case_1 ... ok`.
I have to keyboard interrupt before oom-killer nukes my GUI.
I have 16 GiB of RAM, so low memory shouldn't be the problem.

### Build log

[gist](https://gist.github.com/Quantenzitrone/552012f5d6c8145e82b02ee89c5b88ae)

### Notify maintainers

@06kellyjac 
@evanjs 

### Metadata
```console
[user@system:~]$ nix-info -m
- system: `"x86_64-linux"`
- host os: `Linux 6.2.9-zen1, NixOS, 23.05 (Stoat), 23.05.20230411.3fb8eed`
- multi-user?: `no`
- sandbox: `yes`
- version: `nix-env (Nix) 2.13.3`
- channels(qz): `""`
- channels(root): `""`
- nixpkgs: `not found`
```
I'm on a flake based system, so no channels.

**Comment by 06kellyjac** [comment]
Thanks for raising. It's probably case 4 & 7:

```
test test_package::case_4 has been running for over 60 seconds
test test_package::case_7 has been running for over 60 seconds
```

https://hydra.nixos.org/build/215622104/nixlog/2

Have you tried building the upstream project, both as is & in a nix sandbox? I'm thinking this isn't a nixpkgs specific issue.
If you could raise an issue on the upstream repo I'll update and disable all tests under `test_package` for now

**Comment by Quantenzitrone** [comment]
yep i just set up a registry for the master branch and build the new version with
```sh
nix build master#cargo-geiger --rebuild
```
and no memory overload, so that fixed it
gonna try to build 0.11.5 and 6 without nix now

**Comment by Quantenzitrone** [comment]
building "as is" means just cloning the project and using `cargo build --release` right? or do i have to build on a different disto?

**Comment by 06kellyjac** [comment]
Yeah thats it. Im guessing it will work fine but building the normal project with no networking causes the issue.
